test/resilience/tests/envoygateway.go

```diff
@@ -0,0 +1,242 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+//go:build resilience
+
+package tests
+
+import (
+	"context"
+	"github.com/envoyproxy/gateway/test/resilience/suite"
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/gateway-api/conformance/utils/http"
+	"sigs.k8s.io/gateway-api/conformance/utils/kubernetes"
+	"sigs.k8s.io/gateway-api/conformance/utils/tlog"
+	"testing"
+	"time"
+)
+
+const (
+	namespace    = "envoy-gateway-system"
+	envoygateway = "envoy-gateway"
+	targetString = "successfully acquired lease"
+	apiServerIP  = "10.96.0.1"
+	timeout      = 2 * time.Minute
+	policyName   = "egress-rules"
+	leaseName    = "5b9825d2.gateway.envoyproxy.io"
+	trashHold    = 2
+)
+
+func init() {
+	ResilienceTests = append(ResilienceTests, EGResilience)
+}
+
+var EGResilience = suite.ResilienceTest{
+	ShortName:   "EGResilience",
+	Description: "Envoygateway resilience test",
+	Test: func(t *testing.T, suite *suite.ResilienceTestSuite) {
+		ap := kubernetes.Applier{
+			ManifestFS:     suite.ManifestFS,
+			GatewayClass:   suite.GatewayClassName,
+			ControllerName: "gateway.envoyproxy.io/gatewayclass-controller",
+		}
+		ap.MustApplyWithCleanup(t, suite.Client, suite.TimeoutConfig, "testdata/base.yaml", true)
+
+		//this test will fail until https://github.com/envoyproxy/gateway/pull/4767/files is merged
+		t.Run("Secondary EnvoyGateway instances can serve an up to date xDS", func(t *testing.T) {
+			ctx := context.Background()
+			t.Log("Scaling down the deployment to 0 replicas")
+			err := suite.Kube().ScaleDeploymentAndWait(ctx, envoygateway, namespace, 0, time.Minute, false)
+			require.NoError(t, err, "Failed to scale deployment replicas")
+
+			t.Log("Scaling up the deployment to 3 replicas")
+			err = suite.Kube().ScaleDeploymentAndWait(ctx, envoygateway, namespace, 3, time.Minute, false)
+			require.NoError(t, err, "Failed to scale deployment replicas")
+
+			t.Log("Waiting for leader election")
+			// Ensure leadership was taken
+			name, err := suite.Kube().GetElectedLeader(context.Background(), namespace, leaseName, metav1.Now(), time.Minute*2)
+			require.NoError(t, err, "unable to detect leader election")
+
+			t.Log("Marking the identified pod as leader")
+			suite.Kube().MarkAsLeader(namespace, name)
+
+			t.Log("Simulating API server connection failure for all pods")
+			err = suite.WithResCleanUp(ctx, t, func() (client.Object, error) {
+				return suite.Kube().ManageEgress(ctx, apiServerIP, namespace, policyName, true, map[string]string{})
```

Comment:
This change will allow test Resilience test suite with #4767 pass.  See https://github.com/envoyproxy/gateway/pull/4904

Not sure the exact reason, it seems like a bug from the Calico CNI. 

```suggestion
				return suite.Kube().ManageEgress(ctx, apiServerIP, namespace, policyName, true, map[string]string{"app.kubernetes.io/name": "gateway-helm"})
```

Reply:
hmm i have merged the suggestion,  the Resilience suite still does not pass. 

Did this pass  for you when running locally? It’s possible that delays in resource cleanup caused the tests to behave inconsistently?

Reply:
Understood. Without this fix, the tests fail even with the xDS adjustments. However, this minor change ensures that everything functions as expected. right?

Reply:
@alexwo The Envoy wasn't able to connect to the standby EG instances with the egress rule even though only the API server IP was blocked. Adding a podselector solved that. 

This looks like a Calico bug to me.